### PR TITLE
[RootFS] Update Go compiler shard to 1.16.3

### PR DESCRIPTION
**Pull request by omus**
Was attempting to compile Kubernetes from source and encountered the following:
```
Detected go version: go version go1.13 linux/amd64.
Kubernetes requires go1.15.0 or greater.
Please install go1.15.0 or later.
```
Seems like a good time for an upgrade :)

### CODE REVIEW

File: 0_RootFS/Go/build_tarballs.jl [hunk 1]
```diff
@@ -3,22 +3,24 @@ using BinaryBuilder
...
 
 # Bash recipe for building across all platforms
 script = raw"""
 mv ${WORKSPACE}/srcdir/go ${prefix}/
 """
 
-# We only build for Linux x86_64
+# We only build for host platform: x86_64-linux-musl
 platforms = [
-    # TODO: Switch to musl once https://github.com/rust-lang/rustup.rs/pull/1882 is released
```
Comment (omus) [inline 0_RootFS/Go/build_tarballs.jl:20]: TODO was copy and pasted from the Rust `build_tarballs.jl` file: https://github.com/JuliaPackaging/Yggdrasil/commit/2c40e825c#diff-9d82d1671f46cc0048c4743f450921e16dad3dfc855959d284a5513e25f2d531R9

File: 0_RootFS/Go/build_tarballs.jl [hunk 2]
```diff
@@ -3,22 +3,24 @@ using BinaryBuilder
 include("../common.jl")
 
 name = "Go"
-version = v"1.13"
+version = v"1.16.3"
```
Comment (omus) [inline 0_RootFS/Go/build_tarballs.jl:6]: I'll note that Go seems to treat a minor version change as a major release: https://golang.org/doc/devel/release.html#go1.16.

Comment (giordano) [reply 0_RootFS/Go/build_tarballs.jl:6]: The Go ecosystem has a wonderful versioning scheme :grimacing:

Comment (omus) [reply 0_RootFS/Go/build_tarballs.jl:6]: It does seem though that 1.16 is supposed to be backwards compatible with all version 1 releases

**Comment by omus** [comment]
I'll do a deployment of this shard and validate [HelloWorldGo](https://github.com/JuliaPackaging/Yggdrasil/tree/master/H/HelloWorldGo) works

**Review by giordano — APPROVED** [review]
Looks good to me, as far as I can tell.  Happy to merge after you do your test

**Comment by omus** [comment]
Local build of HelloWorldGo against all targets passed

**Comment by staticfloat** [comment]
Great stuff, @omus!

**Pull request merged**